homeassistant/components/local_calendar/helpers/ics.py

```diff
@@ -0,0 +1,33 @@
+"""ics upload handler."""
+
+from pathlib import Path
+import shutil
+
+from ical.calendar_stream import CalendarStream
+from ical.exceptions import CalendarParseError
+
+from homeassistant.components.file_upload import process_uploaded_file
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from ..const import STORAGE_PATH
+
+
+async def save_uploaded_ics_file(
```

Comment:
I think we should just keep this in the config flow file for now rather than making a separate directory.